### PR TITLE
 Add PropertyManager tests

### PR DESCRIPTION
**Pull request by hughbe**
Extract from #468

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyManagerTests.cs [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+        }
+
+        private class DataSource
+        {
+            public int Property { get; set; }
+        }
+
+        private class SubControl : Control, INotifyPropertyChanged
+        {
+#pragma warning disable 0067
```
Comment (sharwell) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/PropertyManagerTests.cs]: 💡 Don't disable this warning. It is expected to be fixed by using an explicit interface implementation:

```csharp
event PropertyChangedEventHandler INotifyPropertyChanged.PropertyChanged { add { } remove { } }
```
```

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyManager.cs [hunk 2]
```diff
@@ -2,204 +2,183 @@
...
-        }
-
-        private void PropertyChanged(object sender, EventArgs ea) {
+using System.Diagnostics.CodeAnalysis;
+using System.ComponentModel;
+using System.Collections;
+
+namespace System.Windows.Forms
+{
+    public class PropertyManager : BindingManagerBase
```
Comment (sharwell) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyManager.cs]: 📝 I expected this file to not change aside from changes *required* for implementing the tests. As it stands, I no longer know if/how it needed to change to account for testing.

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyManager.cs]: Nothing changed functionally - this is just cleanup in that file (1st commit) tested by the 2nd commit. The 3rd commit avoids an allocatiob

**Comment by zsd4yr** [comment]
@hughbe as always, thanks 😄

**Pull request merged**